subiquity/server/controllers/install.py

```diff
@@ -130,12 +160,97 @@ async def configure_apt(self, *, context):
+                config=step.acquire_config(step, resume_data_file)
+                )
+
+        # Make sure the log directory exists.
+        step.log_file.parent.mkdir(parents=True, exist_ok=True)
+
+        return await run_curtin_command(
+                self.app, context,
+                "install", source,
+                "--set", f'json:stages={json.dumps(step.stages)}',
+                config=str(step.config_file),
+                private_mounts=False)
+
     @with_context(
         description="installing system", level="INFO", childlevel="DEBUG")
     async def curtin_install(self, *, context, source):
-        await run_curtin_command(self.app, context, 'install', source,
-                                 config=self.write_config(),
-                                 private_mounts=False)
+        # Until https://bugs.launchpad.net/curtin/+bug/1876984 gets
+        # fixed, the only way to get curtin to leave the network
+        # config entirely alone is to omit the 'network' stage.
+
+        logs_dir_prefix = Path(
+                self.app.opts.output_base if self.app.opts.dry_run else "/")
+
+        logs_dir = logs_dir_prefix / "var/log/installer/curtin-install"
+        logs_dir.mkdir(parents=True, exist_ok=True)
+
+        # TODO close FD
+        unused, resume_data_path = tempfile.mkstemp()
+        resume_data_file = Path(resume_data_path)
+
+        await self.run_curtin_install_step(CurtinInstallStep(
+                name="initial",
+                stages=[],
+                config_file=logs_dir / "subiquity-initial.conf",
+                log_file=logs_dir / "initial.log",
+                error_file=logs_dir / "initial-error.tar",
+                acquire_config=self.acquire_initial_config,
+            ), resume_data_file=resume_data_file,
+            context=context, source=source)
+
+        generic_steps = [
+            CurtinInstallStep(
+                name="partitioning",
+                stages=["partitioning"],
+                config_file=logs_dir / "subiquity-partitioning.conf",
+                log_file=logs_dir / "partitioning.log",
+                error_file=logs_dir / "partitioning-error.tar",
+                acquire_config=self.acquire_generic_config,
+            ), CurtinInstallStep(
+                name="extract",
+                stages=["extract"],
+                config_file=logs_dir / "subiquity-extract.conf",
+                log_file=logs_dir / "extract.log",
+                error_file=logs_dir / "extract-error.tar",
+                acquire_config=self.acquire_generic_config,
+            ), CurtinInstallStep(
+                name="curthooks",
+                stages=["curthooks"],
+                config_file=logs_dir / "subiquity-curthooks.conf",
+                log_file=logs_dir / "curthooks.log",
+                error_file=logs_dir / "curthooks-error.tar",
+                acquire_config=self.acquire_generic_config,
+            ), CurtinInstallStep(
```

Comment:
I think the hook stage only really does things for installing centos or other non-Ubuntu distros btw.

Reply:
Cool! I did not find too much info about this stage so decided to include it just in case. 
I will drop it if it's not useful for us. Thanks.